### PR TITLE
feature(remote_file): add ability to preserve permissions and ownership

### PR DESCRIPTION
**Pull request by bentsi**
will fix:

```
Sep 08 21:55:50 artifacts-gce-image-jenkins-db-node-10cc16f1-0-1 scylla[6051]:  [shard 0] init - Startup failed: std::filesystem::__cxx11::filesystem_error (error system:13, filesystem error: open failed: Permission denied [/etc/scylla/scylla.yaml])
```

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Review by juliayakovlev — APPROVED** [review]

**Comment by fruch** [comment]
@bentsi check the unit-test, this change seem to be breaking one of them.

**Review by dkropachev — APPROVED** [review]
LGTM

**Review by juliayakovlev — APPROVED** [review]

File: sdcm/remote/remote_file.py [hunk 1]
```diff
@@ -28,9 +28,13 @@ def read_to_stringio(fobj):
 
 
 @contextlib.contextmanager
-def remote_file(remoter, remote_path, serializer=StringIO.getvalue, deserializer=read_to_stringio, sudo=False):
+def remote_file(remoter, remote_path, serializer=StringIO.getvalue, deserializer=read_to_stringio, sudo=False,
+                preserve_ownership=True, preserve_permissions=True):
     filename = os.path.basename(remote_path)
     local_tempfile = os.path.join(tempfile.mkdtemp(prefix='sct'), filename)
+    ownership = remoter.sudo(cmd='stat -c "%U:%G" ' + remote_path).stdout.strip()
```
[the remark this answers is not in the document]
Comment (bentsi) [reply sdcm/remote/remote_file.py]: done

**Comment by fruch** [comment]
@bentsi
any news with this one ?, I want to use the gce image 

BTW, do we have a way to disable the boot ? i.e. utilize user-date in GCE ?

**Comment by bentsi** [comment]
@fruch done. You can set `user-data` in GCE by setting `user-data` key in the Instance Metadata

**Comment by bentsi** [comment]
unrelated unit test failed

**Pull request merged**